src/examples/aes/common/aescircuit.cpp

```diff
@@ -692,6 +709,99 @@ std::vector<uint32_t> AESSBox_Forward_SPLUT(std::vector<uint32_t> input, Boolean
+// tiny-AES-c project, https://github.com/kokke/tiny-AES-c
+// This function produces NB_CONSTANT(AES_ROUNDS+1) round keys in SFE.
+// The round keys are used in each round to decrypt the states. 
+share* BuildKeyExpansion(share* key, BooleanCircuit* circ, bool use_vec_ands) {
+
+	//TODO make the function SIMD-able if multiple keys are needed
+	assert(key->get_nvals() == 1);
+
+	//During the function the representation of the extended key is bytewise
+	std::vector<std::vector<uint32_t>> ex_key_vec(AES_EXP_KEY_BYTES, std::vector<uint32_t>(8));
+
+	uint32_t i, j, k;
+
+	// The first round key is the key itself.
+	{
+		std::vector<uint32_t> key_vec = key->get_wires();
+		for (i = 0; i < key_vec.size(); i++) {
+			ex_key_vec[i / 8][i % 8] = key_vec[i];
+		} 
+	}
+
+	std::vector<uint32_t> s_rc[RCON_SIZE];
+
+	for(i = 0; i < RCON_SIZE; i++) {
+		uint8_t rc_temp = Rcon[i];
+		s_rc[i] = circ->PutCONSGate(rc_temp, 8)->get_wires();
+	}
+	
+	std::vector<std::vector<uint32_t>> tempa(4); // Used for the column/row operations
+
+	// All other round keys are found from the previous round keys.
+  for (i = AES_STATE_COLS; i < NB_CONSTANT * (AES_ROUNDS + 1); ++i) {
+    {
+      k = (i - 1) * 4;
+      tempa[0] = ex_key_vec[k];
+      tempa[1] = ex_key_vec[k + 1];
+      tempa[2] = ex_key_vec[k + 2];
+      tempa[3] = ex_key_vec[k + 3];
+    }
+
+    if (i % AES_STATE_COLS == 0) {
+      // This function shifts the 4 bytes in a word to the left once.
+      // [a0,a1,a2,a3] becomes [a1,a2,a3,a0]
+
+      // Function RotWord()
+      {
+        const std::vector<uint32_t> u8tmp = tempa[0];
+        tempa[0] = tempa[1];
+        tempa[1] = tempa[2];
+        tempa[2] = tempa[3];
+        tempa[3] = u8tmp;
+      }
+
+      // SubWord() is a function that takes a four-byte input word and 
+      // applies the S-box to each of the four bytes to produce an output word.
+
+      // Function Subword()
+      {
+        tempa[0] = PutAESSBoxGate(tempa[0], circ, use_vec_ands);
+				tempa[1] = PutAESSBoxGate(tempa[1], circ, use_vec_ands);
+				tempa[2] = PutAESSBoxGate(tempa[2], circ, use_vec_ands);
+				tempa[3] = PutAESSBoxGate(tempa[3], circ, use_vec_ands);
+      }
+      tempa[0] = circ->PutXORGate(tempa[0], s_rc[i/AES_STATE_COLS]);
+    }
+#if AES_STATE_KEY_BITS == 256
+    if (i % AES_STATE_COLS == 4) {
+      // Function Subword()
+      {
+        tempa[0] = PutAESSBoxGate(tempa[0], circ, use_vec_ands);
+				tempa[1] = PutAESSBoxGate(tempa[1], circ, use_vec_ands);
+				tempa[2] = PutAESSBoxGate(tempa[2], circ, use_vec_ands);
+				tempa[3] = PutAESSBoxGate(tempa[3], circ, use_vec_ands);
+      }
+    }
+#endif
+		k = (i - AES_STATE_COLS) * 4;
+		j = i * 4;
+    ex_key_vec[j] = circ->PutXORGate(ex_key_vec[k], tempa[0]);
+    ex_key_vec[j + 1] = circ->PutXORGate(ex_key_vec[k + 1], tempa[1]);
+    ex_key_vec[j + 2] = circ->PutXORGate(ex_key_vec[k + 2], tempa[2]);
+    ex_key_vec[j + 3] = circ->PutXORGate(ex_key_vec[k + 3], tempa[3]);
+  }
```

Comment:
Use tabs for indentation for sake of consistency.

Reply:
Ops, I missed that. I will change it.